tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/base.tsx

```diff
@@ -484,3 +484,22 @@ export const FeatureValueInput = (
     />
   );
 };
+
+export type FeatureShortTextData = {
+  maximum_length: number;
+};
+
+export const FeatureShortTextInput = (props: FeatureValueProps<string, string, FeatureShortTextData>) => {
+  if (!props.serverData) {
+    return <Box>Loading...</Box>;
+  }
+
+  return (
+    <Input
+      width="100%"
+      value={props.value}
+      maxLength={props.serverData.maximum_length}
```

Comment:
This should be in the regular text input, since the max length is on every string field.

Reply:
Sorry for the late reply, I'm not too familiar with TypeScript but by this I assume the maximum_length is to be added into the TextInput too? I'll test this tomorrow but I believe the maximum limit is meant to be short.

Reply:
There is no need to create a "ShortTextInput" component when every preference already sends a max size if it's text. So my point is just remove this new component and put it on the normal text input

Reply:
After checking and testing it, it seems that regular TextInput does not appear to make use of the max size which allows me to bypass the max size and spawn in with a really long ringtone. I believe this component will still be needed for a proper input limit.

Reply:
Yes. My point is you should **add** it to the regular textinput, because you're already adding it to everything's server data. Making a redundant component when you can just add the functionality to the base component would be completely fine.

There is already a specific FeatureTextInput component or something like that that is used by every preference that is just text.